### PR TITLE
use new_tibble() instead of as_tibble()

### PR DESCRIPTION
**Pull request by romainfrancois**
This limits a bit further the `across()` footprint compared to #4982 

image

This perhaps needs some extra work to : 
- do the name repair thing once, perhaps in the `setup` function. 
- perhaps check that all the results have the same size: 

``` r
str(tibble::new_tibble(list(x = 1:2, y = 1:3), nrow = 3))
#> Classes 'tbl_df', 'tbl' and 'data.frame':    3 obs. of  2 variables:
#>  $ x: int  1 2
#>  $ y: int  1 2 3
```

<sup>Created on 2020-03-17 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

**Comment by DavisVaughan** [comment]
I thought about this too, but didn't pull the trigger because I wasn't sure how much safety we needed in `across()`, since the output all comes from the user

> perhaps check that all the results have the same size:

Or are they recycled to a common size? I think that is what `as_tibble.list` would do. `vec_recycle_common()` could be used (it would error on non-recyclable sizes for us).

**Comment by DavisVaughan** [comment]
For name repair, yea I guess we could use `vctrs::vec_as_names(names, repair = "check_unique")` in the setup? I think that matches the default of `as_tibble.list()`

**Comment by romainfrancois** [comment]
I've tried  `as_tibble(.name_repair = identity)` but that gives this: 

image

so yeah I suppose at this point it becomes an issue about `tibble:::as_tibble.list()` unless we're happy about what `new_tibble()` does, but that's a bit risky perhaps.

**Comment by krlmlr** [comment]
I'm not sure we can make `as_tibble()` much faster (it involves S3 dispatch to begin with); being fast is `new_tibble()`'s job. There's `validate_tibble()` that can be used for checking, we can enhance in tibble 3.0.0 to check only a subset of the properties.

**Comment by krlmlr** [comment]
Also, would we rather use `rlang::new_data_frame()` ?

**Comment by romainfrancois** [comment]
I don't see it, is that in a PR ?

**Comment by DavisVaughan** [comment]
I imagine he means `vctrs::new_data_frame()`. Since we pass this data frame back to the user, I would expect it to be a tibble

File: R/across.R [hunk 1]
```diff
@@ -125,9 +125,9 @@ across <- function(.cols = everything(), .fns = NULL, ..., .names = NULL) {
     }
   }
 
+  out <- vec_recycle_common(!!!out)
   names(out) <- names
-
-  as_tibble(out)
+  new_tibble(out, nrow = vec_size(out[[1]]))
```
Comment (DavisVaughan) [inline R/across.R]: How about this?

```
size <- vec_size_common(!!!out)
out <- vec_recycle_common(!!!out, .size = size)

...

new_tibble(out, nrow = size)
```

**Pull request merged**